### ISSUE TITLE
matches = truths[best_truth_idx]

### ISSUE THREAD

**Issue by DonghoonPark12** [amdegroot/ssd.pytorch#404]
Damn. Is there any one who can clearly explain below two lines???
[link](https://github.com/amdegroot/ssd.pytorch/blob/5b0b77faa955c1917b0c710d770739ba8fbff9b7/layers/box_utils.py#L107)
```python
matches = truths[best_truth_idx]          # Shape: [num_priors,4]
conf = labels[best_truth_idx] + 1         # Shape: [num_priors]
```

**Comment by JappaB** [comment]
@DonghunP The +1 is probably because the class indices are zero-indexed, while at the same time, the output of the neural network has the background at index zero.

**Comment by KSeangTan** [comment]
@DonghunP 
First line is to find out what ground truth of each prior box belong to
For example, if you have two objects (2 ground truth), and 3 prior boxes, your best_truth_idx = [0, 1, 1]. 
It means that first prior box best matching with first ground truth, second prior box best match with second ground truth, and so third prior box.
For second line, it returns what class of each prior box represents, +1 is due to we have one additional background label.
